git-config/src/file/git_config.rs

```diff
@@ -145,28 +172,70 @@ impl<'event> GitConfig<'event> {
     ///
     /// [`git-config`'s documentation]: https://git-scm.com/docs/git-config#Documentation/git-config.txt-FILES
     #[inline]
-    pub fn from_paths(paths: &[&Path]) -> Result<Self, ParserOrIoError<'static>> {
+    pub fn from_paths(paths: &[&Path], git_install_dir: Option<&std::path::Path>) -> Result<Self, from_paths::Error> {
```

Comment:
I think it's always good to 'not lie' about the inputs, so if a `Vec<PathBuf>` is needed, it's better to have people pass exactly that. That way, if the caller happens to have that, it can just pass it on without extra copies being created unnecessarily.
It's just an idea, and maybe it's not the right thing to do here.